compiler/passes/normalize.cpp

```diff
@@ -1339,71 +1339,53 @@ static void init_typed_var(VarSymbol* var,
     return;
   }
 
-  //
-  // initialize variable based on specified type and then assign it
-  // the initialization expression if it exists
-  //
-  bool isNoinit     = init && init->isNoInitExpr();
-  bool moduleNoinit = false;
-
-  if (isNoinit && !fUseNoinit) {
-    // In the case where --no-use-noinit is thrown, we want to still use
-    // noinit in the module code (as the correct operation of complexes
-    // depends on it).
-
-    // Lydia note: The requirement for complexes is expected to go away when
-    // we transition to constructors for all types instead of the _defaultOf
-    // function
-    Symbol* moduleSource = var;
+  if (init->isNoInitExpr() == true) {
```

Comment:
If Coverity were still useful, it would probably yell at you for accessing a method on something that could be NULL.  I'm not sure if you change the `hasFlag(FLAG_PARAM)` case above this code to no longer check that init is not NULL, but either way, I think it might be good to assert at the beginning of this function that init is not NULL (with maybe a comment pointing to the other version), in case someone comes along later and mistakenly tries to use this function instead of the one for init being NULL.  Doing that will ensure that your assumptions and preconditions will continue to be met going forward.